src/Microsoft.Sbom.Api/Workflows/SBOMGenerationWorkflow.cs

```diff
@@ -109,6 +112,14 @@ await using (SBOMConfigs.StartJsonSerializationAsync())
 
                     return !validErrors.Any();
                 }
+                catch (ManifestFolderExistsException e)
+                {
+                    Recorder.RecordException(e);
```

Comment:
instead of having duplicate code from below, maybe in the generic exception, check if exception is type `ManifestFolderExistsException` and then update `deleteSBOMDir`

pseudo code
```
if (e is not ManifestFolderExistsException)
{
   deleteSBOMDir = true
}
```